openedx/features/calendar_sync/plugins.py

```diff
@@ -0,0 +1,82 @@
+"""
+Platform plugins to support Calendar Sync toggle.
+"""
+
+
+from django.urls import reverse
+from django.utils.translation import ugettext as _
+
+from lms.djangoapps.courseware.permissions import CALENDAR_SYNC
+from openedx.features.calendar_sync.api import SUBSCRIBE, UNSUBSCRIBE
+from openedx.features.calendar_sync.models import UserCalendarSyncConfig
+from openedx.features.course_experience import CALENDAR_SYNC_FLAG, RELATIVE_DATES_FLAG
+from openedx.features.course_experience.course_tools import CourseTool
+from student.models import CourseEnrollment
+
+
+class CalendarSyncToggleTool(CourseTool):
+    """
+    The Calendar Sync toggle tool.
+    """
+    link_title = _('Calendar Sync')
+    toggle_data = {'toggle_data': ''}
+
+    @classmethod
+    def analytics_id(cls):
+        """
+        Returns an id to uniquely identify this tool in analytics events.
+        """
+        return 'edx.calendar-sync'
+
+    @classmethod
+    def is_enabled(cls, request, course_key):
+        """
+        The Calendar Sync toggle tool is limited to user enabled through a waffle flag.
+        Staff always has access.
+        """
+        if CALENDAR_SYNC_FLAG.is_enabled(course_key) and RELATIVE_DATES_FLAG.is_enabled(course_key):
+            if (request.user.has_perm(CALENDAR_SYNC, course_key) or
+                    CourseEnrollment.is_enrolled(request.user, course_key)):
```

Comment:
Both if statements can be combined

Reply:
I'm opting to keep these separate for clarity on which flags we are gating on and what the standard logic is.

Reply:
Is the perm logic temporary for testing? Why not just simply check enrollment? Feels weird to press the button outside that context.

Reply:
I guess I added it in for staff testing, but agree that it could be a bit weird for an unenrolled staff member to click it. But it does at least show staff that it shows up which might be good? I'm not sure what to do for this situation

Reply:
I mean, staff can always enroll if they want to see it. I'm leery of things that show a different experience for staff, as it might actually make it harder to test.

How does a staff know it doesn't show up for everyone? If they aren't looking at the code, they might be worried it is, when they see it. People like UX folk wouldn't know it isn't normally there.

Reply:
Yeah, that's fair. It's the logic the bookmarks link was using too, but I'll remove it for the calendar at least.